### ISSUE TITLE
One driver per browser pool instance

### ISSUE THREAD

**Issue by szmarczak** [apify/browser-pool#58]
As per https://github.com/apify/apify-js/pull/1228#discussion_r744304601

**Comment by mnmkng** [comment]
I assume you mean one type of driver (plugin), right? E.g. using 3 plugins to rotate chrome + firefox + webkit in Playwright still feels like a legit use-case to me.

**Comment by szmarczak** [comment]
Correct, the API remains the same. However mixing Playwright with Puppeteer makes it hard to use.

**Comment by mnmkng** [comment]
Yeah. I think it would make sense. You can always create two pools when you need to manage two libraries. What do you think @petrpatek ?

**Comment by petrpatek** [comment]
Yeah, I think it makes sense to mix only multiple browsers, but I can't see any reason to use playwright and puppeteer simultaneously.

**Comment by mnmkng** [comment]
Ok, let's do it then 👍🏻

**Comment by mnmkng** [comment]
@B4nan I think this would make sense to do before v3

**Comment by B4nan** [comment]
> I assume you mean one type of driver (plugin), right? E.g. using 3 plugins to rotate chrome + firefox + webkit in Playwright still feels like a legit use-case to me.

I am a bit lost here, what is _driver_, if its plugin, how can we allow only one type per pool but still allow using 3 different plugins? Trying to read the conversion but I see quite a lot of contradictions.

**Comment by mnmkng** [comment]
Basically that we wouldnt allow combining Puppeteer and Playwright plugins in a single browser pool. It's a breaking change, but it shouldnt affect anything on our end.

**Comment by B4nan** [comment]
But how can you then achieve "using 3 plugins to rotate chrome + firefox + webkit in Playwright"?

**Comment by B4nan** [comment]
Oh right, so you just want to allow _only playwright_, not mixing it with puppeteer

**Comment by mnmkng** [comment]
Validation, a few changes in docs, plus I guess the TypeScript types might require some tweaks? Dunno how dynamic they are.

**Comment by mnmkng** [comment]
@B4nan ping
